tests/functional/testHeartbeat.js

```diff
@@ -54,21 +49,30 @@ describe('Backend failure', function() {
     await vpn.waitForQuery(
         queries.screenInitialize.AUTHENTICATE_VIEW.visible());
 
-    await backendFailureAndRestore();
+    await vpn.forceHeartbeatFailure();
+
+    // TODO: without call to click on cancel this test fails because 
+    // we get stuck on the loading spinner 
+    // after authenticating following the changes in mozillavpn.cpp
+    // to remove the call to backend failure in heartbeatCompleted().
+    // Is clicking on "cancel" here okay?
+    await vpn.waitForQueryAndClick(
+      queries.screenAuthenticating.CANCEL_FOOTER_LINK.visible());
+
```

Comment:
Maybe the solution is to raise the backend failure only on StateAuthenticating, because it does make sense to raise a Guardian failure during authentication. So, something like:
void MozillaVPN::heartbeatCompleted(bool success) {
  logger.debug() << "Server-side check done:" << success;

  if (!success) {
    if (state() == StateAuthenticating) {
        m_private->m_connectionManager.backendFailure();
    }
    return;
  }